sympy/physics/mechanics/tests/test_joint.py

```diff
@@ -0,0 +1,347 @@
+from sympy import Symbol, sin, cos, ImmutableMatrix as Matrix
+from sympy.physics.vector import Vector, cross
+from sympy.physics.mechanics import (dynamicsymbols, Body, Joint, PinJoint,
+                                     SlidingJoint, CylindricalJoint, PlanarJoint,
+                                     SphericalJoint)
+from sympy.utilities.pytest import raises
+
+
+def test_joint_abstract_class():
+    parent = Body('parent')
+    child = Body('child')
+    raises(NotImplementedError, lambda: Joint('joint', parent, child))
+
+
+def test_pin_joint():
+    parent = Body('parent')
+    child = Body('child')
+    l = Symbol('l')
+    pin_joint = PinJoint('pin_joint', parent, child, child_point_pos=(0, l, 0))
+
+    assert pin_joint.name == 'pin_joint'
+    assert pin_joint.parent == parent
+    assert pin_joint.child == child
+    assert pin_joint._parent_joint_location == Vector(0)
+    assert pin_joint._child_joint_location == l * child.frame.y
+    assert pin_joint.parent_axis == parent.frame.x
+    assert pin_joint.child_axis == child.frame.x
+
+    theta = dynamicsymbols('pin_joint_theta')
+    thetad = dynamicsymbols('pin_joint_theta', 1)
+    omega = dynamicsymbols('pin_joint_omega')
```

Comment:
And as you have here, you'll need unique names among joint types. So if you use `theta` in the pin joint and cylindrical joint, for example, and you have both joints in a system, then they need to be unique names.

If you made a class attribute on the Joint class for creating unique names then it would work for all joints.